core/src/main/scala/kafka/cluster/Partition.scala

```diff
@@ -298,7 +300,8 @@ class Partition(val topic: String,
 
           // check if the HW of the partition can now be incremented
           // since the replica maybe now be in the ISR and its LEO has just incremented
-          maybeIncrementLeaderHW(leaderReplica)
+          // TODO: is this maybeIncrementLeaderHW() necessary?
```

Comment:
I think we need this since every time the follower's fetch offset advances, we may need to advance the HW.